backend/tester/src/test/scala/io/theorg/tester/OrgchartApiTestSuite.scala

```diff
@@ -0,0 +1,136 @@
+package io.theorg.tester
+
+import caliban.client.Operations.{RootMutation, RootQuery}
+import caliban.client.SelectionBuilder
+import sttp.capabilities
+import sttp.capabilities.WebSockets
+import sttp.capabilities.zio.ZioStreams
+import sttp.client3.SttpBackend
+import sttp.model.Uri
+import zio.test.Assertion._
+import zio.test.TestAspect.sequential
+import zio.test._
+import zio.test.environment.TestEnvironment
+import zio.{Managed, Task, ZIO}
+import io.theorg.tester._
+import io.theorg.tester.client.backend._
+import java.util.UUID
+import sttp.client3.httpclient.zio.HttpClientZioBackend
+import zio.ZManaged
+import zio.Ref
+
+object OrgchartApiTestSuite extends DefaultRunnableSpec {
+
+  private val MaxNodesPerLevel = 3
+  private val AddNodeConcurrency = 4
+  private val MaxAddNodeRequests = 1000
+  private val targetGraphqlUrl = "http://localhost:8080/graphql"
+
+  final case class OrgChartNodeMapped(
+      id: UUID,
+      title: String,
+      parentId: Option[UUID]
+  )
+  private val apiUrl: Uri =
+    Uri.parse(targetGraphqlUrl).toOption.get
+
+  def withServer(
+      test: SttpBackend[Task, ZioStreams with WebSockets] => ZIO[
+        TestEnvironment,
+        Throwable,
+        TestResult
+      ]
+  ): ZIO[TestEnvironment, Throwable, TestResult] = {
+    val testJob =
+      for {
+        backend <- HttpClientZioBackend.managed()
+        response <- Managed.fromEffect(test(backend))
+      } yield response
+
+    testJob.useNow
+  }
+
+  val orgChartSelection = OrgChart.nodes(
+    (OrgChartNode.id.map(
+      UUID.fromString
+    ) ~ OrgChartNode.title ~ OrgChartNode.parentId.map(
+      (_.map(UUID.fromString))
+    )).mapN(OrgChartNodeMapped(_, _, _))
+  )
+
+  def getNodes(level: Int = 0) = Query.loadOrgChart(level)(
+    orgChartSelection
+  )
+
+  def addNode(title: String, parentId: Option[UUID]) = {
+    Mutation.addNode(AddNodeInput(title, parentId.map(_.toString())))(
+      AddNodeResult.orgChart(orgChartSelection)
+    )
+  }
+
+  private val calibanServerSpec =
+    suite("Caliban server Spec")(
+      testM("Empty on first run") {
+        withServer { backend =>
+          val response = getNodes().toRequest(apiUrl).send(backend)
+          assertM(response.map(_.code.code))(equalTo(200))
+          assertM(response.map(_.body).absolve.map(_.flatten.map(_.size)))(
+            equalTo(Some(0))
+          )
+        }
+      },
```

Comment:
Nice. Maybe extract 200 to HttpOkStatusCode constant. 
I suppose the monadic asserts allow you to assert on the payload contained inside monadic effects?
Is there a way to avoid the `.code.code` by using a monad transformer?

Reply:
.code is a strong typed value so .code just gets the inner value, it's not monadic